cockatrice/src/player.cpp

```diff
@@ -1076,6 +1076,25 @@ void Player::actCreatePredefinedToken()
     actCreateAnotherToken();
 }
 
+void Player::actCreateRelatedCard()
+{
+    // get he target card name
+    QAction *action = static_cast<QAction *>(sender());
+    CardInfo *cardInfo = db->getCard(action->text());
+
+    // create the token for the related card
+    Command_CreateToken cmd;
+    cmd.set_zone("table");
+    cmd.set_card_name(cardInfo->getName().toStdString());
+    cmd.set_color(cardInfo->getColors().isEmpty() ? QString().toStdString() : cardInfo->getColors().first().toLower().toStdString());
+    cmd.set_pt(cardInfo->getPowTough().toStdString());
+    cmd.set_destroy_on_zone_change(true);
+    cmd.set_x(-1);
+    cmd.set_y(0);
```

Comment:
Instead of setting x and y by default, why not acquire where the card the transformation was called from (i.e. Get Delver of Secret's location) and place this new card right on top of it (giving it the same X/Y of Delver)